Fastpack/Cache.rei

```diff
@@ -0,0 +1,27 @@
+type t
+type load =
+  | Empty
+  | Load(params)
+and params = {
+  currentDir: string,
+  projectRootDir: string,
+  mock: list((string, Config.Mock.t)),
+  nodeModulesPaths: list(string),
+  resolveExtension: list(string),
+  preprocess: list(Preprocessor.config),
+};
+
+let make: load => Lwt.t(t);
```

Comment:
no need to introduce new type just to describe two behaviours of creating a cache, instead let's have two constructor functions:
```
let makeFresh : unit => Lwt.t(t);
let makeOfProject : (~currentDir: string, ...) => Lwt.t(t);
```

That way we get rid of `load` and `params` types which are not really needed as we are not going to store refs to them at all (they are only used to construct cache).

Reply:
Agreed. I like the idea